dedalus/core/solvers.py

```diff
@@ -133,8 +139,23 @@ def solve_sparse(self, pencil, N, target, rebuild_coeffs=False, **kw):
         # Solve as sparse general eigenvalue problem
         A = pencil.L_exp
         B = -pencil.M_exp
+        # Solve for the right eigenvectors
         self.eigenvalues, self.eigenvectors = scipy_sparse_eigs(A=A, B=B, N=N, target=target, matsolver=self.matsolver, **kw)
+        if left == True:
+            # Solve for the left eigenvectors
+            #TODO: I think A and B are sparse.csr_matrix objects, in which case csr_matrix.getH might be better than np.conjugate(np.transpose()). Need to test.
+            self.left_eigenvalues, self.left_eigenvectors = scipy_sparse_eigs(A=np.conjugate(np.transpose(A)),
+                                                                              B=np.conjugate(np.transpose(B)),
+                                                                              N=N, target=np.conjugate(target),
+                                                                              matsolver=self.matsolver, **kw)
+            # The following isn't an error, just bad luck: sometimes the sparse solver doesn't hit the same targets for left as it does for right eigenvectors
```

Comment:
Maybe we should raise here?  Or only return the subset of modes that match?

Reply:
Also is this mismatch ever just a sorting issue?  Maybe we need to compare after sorting.

Reply:
It's such a rare occurrence that I don't have the clearest recollection of the details, but from what I do remember, it's not just a sorting issue. I wish I had saved the example where this happened. I think raising is a reasonable solution except I can imagine at least one edge case where I wouldn't care about the error: suppose I want to use just one particular left eigenvector to calculate something, and I find that setting some particular target and Nev=3, I consistently find the correct left eigenvector as the third mode, whereas the first two modes are junk/mismatches that I don't care about. On the one hand, not raising risks misleading some users; on the other hand, raising means in that edge case I would have to keep fishing around to avoid raising the error.

Reply:
It seems like a good option might be to raise on default, and add a keyword option to disable this and warn instead, so you can continue if you're aware of this issue and don't mind?

Reply:
Great idea! I'll work on implementing that.

Reply:
Now raises unless keyword `raise_on_mismatch = False`. Should it also sort the modes & eigenvalues before this point?